pkg/ingester/ingester.go

```diff
@@ -208,7 +209,13 @@ func New(cfg Config, clientConfig client.Config, store ChunkStore, limits *valid
 
 	if cfg.WAL.Enabled {
 		if err := os.MkdirAll(cfg.WAL.Dir, os.ModePerm); err != nil {
-			return nil, err
+			// Best effort try to make path absolute for easier debugging.
+			path, _ := filepath.Abs(cfg.WAL.Dir)
+			if path == "" {
+				path = cfg.WAL.Dir
+			}
+
+			return nil, fmt.Errorf("creating WAL folder at %q: %w", path, err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("failed creating WAL folder at %q: %w", path, err)
```

Reply:
From https://github.com/grafana/loki/pull/4705#discussion_r744882151

>Repeating error keyword in wrapped error message is often undesired, as it creates a lot of repetition both in code and for the reader when reported error is deeply nested.

Reply:
Thanks for providing the context! 👍